tokio/tests/time_sleep.rs

```diff
@@ -317,15 +317,15 @@ async fn drop_after_reschedule_at_new_scheduled_time() {
 
```

Comment:
We should have a test that asserts that `Sleep` is `!Unpin` (same for other types that contain pinned `Sleep`s).

Reply:
Such a test can easily be constructed using the macros in `tokio/tokio/tests/async_send_sync.rs`.